Sources/swift-container-plugin/Documentation.docc/authentication.md

```diff
@@ -6,12 +6,95 @@ Configure the plugin to authenticate to your container registry.
 
 The following example shows placeholder values for the registry `registry.example.com`:
 
 ```
 machine registry.example.com
   login myuser
   password mypassword
-```
\ No newline at end of file
+```
+
+The following examples show how to set up the plugin for some popular registry providers.
+
+### Docker Hub
+
+> Don't use your Docker Hub account password to push and pull images.
+> Personal Access Tokens have restricted privileges and you can create a separate token for each integration you use, which you can monitor independently and revoke at any time.
+
+1. Log into Docker Hub and [generate a Personal Access Token](https://docs.docker.com/security/for-developers/access-tokens/) for Swift Container Plugin.
+
+2. **Set the token's access permissions to *Read & Write*.**
+
+3. Copy the token and add it, together with your Docker ID, to your `.netrc` file under the machine name `auth.docker.io`:
+
+The final `.netrc` entry should be similar to this:
+
+```
+machine auth.docker.io
+  login mydockerid
+  password dckr_pat_B3FwrU...
+```
+
+### GitHub Container Registry
+
+> GitHub Container Registry only supports authentication using a [Personal Access Token (classic)](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry).
+> A fine-grained personal access token cannot be used.
+
+1. Log into GitHub and [generate a Personal Access Token (classic)](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic) for Swift Container Plugin.
+
+2. **Select the *write:packages* scope.**
+
+3. Copy the token and add it, together with your GitHub username, to your `.netrc` file:
+
+The final `.netrc` entry should be similar to this:
+
+```
+machine ghcr.io
+  login mygithubusername
+  password ghp_fAOsWl...
+```
+
+### Amazon Elastic Container Registry
+
+> Amazon Elastic Container Registry uses [short-lived authorization tokens](https://docs.aws.amazon.com/AmazonECR/latest/userguide/registry_auth.html#registry-auth-token) which expire after 12 hours.
+>
+> To generate an ECR authentication token, you must [first install the AWS CLI tools.](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html)
+
+1. Use the `aws` CLI tool to [generate an authentication token](https://docs.aws.amazon.com/AmazonECR/latest/userguide/registry_auth.html#registry-auth-token).
+You'll need to know the name of the [AWS region](https://docs.aws.amazon.com/global-infrastructure/latest/regions/aws-regions.html) in which your registry is hosted.
+Registries in different AWS regions are separate and require different authentication tokens.
+
+    For example, the following command generates a token for ECR in the `us-west-2` region:
```

Comment:
```suggestion
For example, the following command generates a token for ECR in the `us-west-2` region:
```

Reply:
Same as below - I'm trying to indent these lines so that they stay under the bulleted list.